pkg/stanza/operator/input/windows/event.go

```diff
@@ -18,54 +18,64 @@ type Event struct {
 	handle uintptr
 }
 
+// NewEvent will create a new event from an event handle.
+func NewEvent(handle uintptr) Event {
+	return Event{
+		handle: handle,
+	}
+}
+
 // RenderSimple will render the event as EventXML without formatted info.
-func (e *Event) RenderSimple(buffer Buffer) (EventXML, error) {
+func (e *Event) RenderSimple(buffer Buffer) (*EventXML, error) {
 	if e.handle == 0 {
-		return EventXML{}, fmt.Errorf("event handle does not exist")
+		return nil, fmt.Errorf("event handle does not exist")
 	}
 
 	bufferUsed, err := evtRender(0, e.handle, EvtRenderEventXML, buffer.SizeBytes(), buffer.FirstByte())
-	if errors.Is(err, ErrorInsufficientBuffer) {
-		buffer.UpdateSizeBytes(*bufferUsed)
-		return e.RenderSimple(buffer)
-	}
-
 	if err != nil {
```

Comment:
One neat observation: this now doesn't follow the same pattern as in `RenderDeep` both seem to work. Any reason to prefer one pattern over the other?

Reply:
True, I added the `err != nil` first, thinking it was a bit clearer than checking the type of error before knowing whether there even is one to begin with. I think the behavior is effectively the same but will update `RenderDeep` to match the pattern.

Reply:
This is now the same between RenderSimple and RenderDeep